packages/jest-resolve/src/__tests__/resolve.test.ts

```diff
@@ -17,8 +18,23 @@ import type {ResolverConfig} from '../types';
 
 jest.mock('../__mocks__/userResolver');
 
+// Do not fully mock `resolve` because it is used by Jest. Doing it will crash
+// in very strange ways. Instead just spy on the method `sync`.
+jest.mock('resolve', () => {
+  const originalModule = jest.requireActual('resolve');
+  return {
+    ...originalModule,
+    sync: jest.spyOn(originalModule, 'sync'),
+  };
+});
+
+const mockResolveSync = <
```

Comment:
Full disclosure: I've never used Jest+TS, I don't know if there is a better way of doing this.